src/it/scala/io/iohk/ethereum/sync/util/RegularSyncItSpecUtils.scala

```diff
@@ -0,0 +1,468 @@
+import io.iohk.ethereum.blockchain.sync.regular.{BlockBroadcasterActor, RegularSync}
+import io.iohk.ethereum.blockchain.sync.{BlockBroadcast, BlockchainHostActor, PeersClient, TestSyncConfig}
+import io.iohk.ethereum.consensus.ethash.{EthashConfig, EthashConsensus}
+import io.iohk.ethereum.consensus.{ConsensusConfig, FullConsensusConfig, ethash}
+import io.iohk.ethereum.crypto.kec256
+import io.iohk.ethereum.db.components.{RocksDbDataSourceComponent, Storages}
+import io.iohk.ethereum.db.dataSource.{RocksDbConfig, RocksDbDataSource}
+import io.iohk.ethereum.db.storage.pruning.{ArchivePruning, PruningMode}
+import io.iohk.ethereum.db.storage.{AppStateStorage, Namespaces}
+import io.iohk.ethereum.domain._
+import io.iohk.ethereum.ledger._
+import io.iohk.ethereum.mpt.{HashNode, MerklePatriciaTrie, MptNode, MptTraversals}
+import io.iohk.ethereum.network.EtcPeerManagerActor.PeerInfo
+import io.iohk.ethereum.network.PeerManagerActor.{FastSyncHostConfiguration, PeerConfiguration}
+import io.iohk.ethereum.network.discovery.Node
+import io.iohk.ethereum.network.discovery.PeerDiscoveryManager.{DiscoveredNodesInfo, DiscoveryNodeInfo}
+import io.iohk.ethereum.network.handshaker.{EtcHandshaker, EtcHandshakerConfiguration, Handshaker}
+import io.iohk.ethereum.network.p2p.EthereumMessageDecoder
+import io.iohk.ethereum.network.p2p.messages.CommonMessages.NewBlock
+import io.iohk.ethereum.network.rlpx.AuthHandshaker
+import io.iohk.ethereum.network.rlpx.RLPxConnectionHandler.RLPxConfiguration
+import io.iohk.ethereum.network.{EtcPeerManagerActor, ForkResolver, KnownNodesManager, PeerEventBusActor, PeerManagerActor, ServerActor}
+import io.iohk.ethereum.nodebuilder.{PruningConfigBuilder, SecureRandomBuilder, ShutdownHookBuilder, VmSetup}
+import io.iohk.ethereum.ommers.OmmersPool
+import io.iohk.ethereum.sync.util.SyncCommonItSpec._
+import io.iohk.ethereum.sync.util.SyncCommonItSpecUtils.FakePeerCustomConfig.defaultConfig
+import io.iohk.ethereum.sync.util.SyncCommonItSpecUtils._
+import io.iohk.ethereum.transactions.PendingTransactionsManager
+import io.iohk.ethereum.utils.ServerStatus.Listening
+import io.iohk.ethereum.utils._
+import io.iohk.ethereum.vm.EvmConfig
+import io.iohk.ethereum.{Fixtures, Timeouts}
+import monix.eval.Task
+
+import scala.annotation.tailrec
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.{FiniteDuration, _}
+import scala.util.Try
+object RegularSyncItSpecUtils {
+
+  object ShutdownHookBuilder extends ShutdownHookBuilder with Logger
+  class ValidatorsExecutorAlwaysSucceed extends MockValidatorsAlwaysSucceed {
+    override def validateBlockAfterExecution(block: Block, stateRootHash: ByteString, receipts: Seq[Receipt], gasUsed: BigInt)
+    : Either[BlockExecutionError, BlockExecutionSuccess] = Right(BlockExecutionSuccess)
+  }
+
+  object ValidatorsExecutorAlwaysSucceed extends ValidatorsExecutorAlwaysSucceed
+
+  class FakePeer(peerName: String, fakePeerCustomConfig: FakePeerCustomConfig)
```

Comment:
to be hones regular sync `FakePeer` looks almost the same as  `FastSyncItSpecUtils.FakePeer` (except some regular sync specific clasess) maybe we could have only one `FakePeer` in some kinda `SyncItSpecUtils` and in regular sync tests start it by `startRegularSync` and in fast sync tests start it by `startFastSync` ?

Reply:
Add new abstract class called “CommonFakePeer”